txnkv/transaction/2pc.go

```diff
@@ -252,6 +261,12 @@ func (m *memBufferMutations) Push(op kvrpcpb.Op, isPessimisticLock, assertExist,
 	}
 	handle.UserData = aux
 	m.handles = append(m.handles, handle)
+	if value != nil {
+		if m.overlay == nil {
+			m.overlay = make(map[unionstore.MemKeyHandle][]byte)
+		}
+		m.overlay[handle] = value
```

Comment:
Maybe we could assert `isPessimisticLock == true` here?

Reply:
Do you mean panic on assertion failed?

Reply:
Yes, as it may prevent some data inconsistency risks.